scipy/stats/tests/test_morestats.py

```diff
@@ -1813,6 +1813,12 @@ def test_gh_6873(self):
         # The expected value was computed with mpmath.
         assert_allclose(llf, -17.93934208579061)
 
+    def test_instability_gh20021(self):
+        data = [2003, 1950, 1997, 2000, 2009]
+        llf = stats.boxcox_llf(1e-8, data)
+        # The expected value was computed with mpmath.
```

Comment:
I see that this was following the example above, but we now try to include enough information to reproduce reference values. Without it, it's harder for me to check the value so I can sign off on it, and it's even tougher for someone else who's interested in the future.

It's OK with me to use `mpsci` here, but can you include the essential informaiton, like:
```
# The expected value was computed with mpsci
# mpsci.stats.boxcox_llf(1e-8, data)
```
and maybe the `mpmath.mp.dps` value, if you set it? When I use 100 digits of precision, I got a different value with:

<details>

```python3
from mpmath import mp
mp.dps = 100

def var(x, ddof=0):
    """
    Variance of the values in the sequence x.
    """
    n = len(x)
    with mp.extraprec(16):
        sumx = mp.fsum(x)
        meanx = sumx / n
        varx = mp.fsum((mp.mpf(t) - meanx)**2 for t in x)/(n - ddof)
    return varx

def boxcox_llf(lam, x):
    """
    Log-likelihood for maximum likelihood estimation of the Box-Cox parameters.
    """
    with mp.extradps(5):
        lam = mp.mpf(lam)
        x = [mp.mpf(t) for t in x]
        n = len(x)

        logdata = [mp.log(t) for t in x]
        sumlogdata = mp.fsum(logdata)

        # Compute the variance of the transformed data.
        if lam == 0:
            variance = var(logdata)
        else:
            # Transform without the constant offset 1/lam.  The offset does
            # not effect the variance, and the subtraction of the offset can
            # lead to loss of precision.
            variance = var([t**lam / lam for t in x])

        return (lam - 1) * sumlogdata - n/2 * mp.log(variance)

data = [2003, 1950, 1997, 2000, 2009]
boxcox_llf(mp.mpf(1e-8), data)  # -15.3240127286901659
```

</details>

Reply:
Thanks for your suggestion, I add some comment and update the math above.

Reply:
Thanks!